### DISCUSSION TITLE
I have trouble with a transformed default value

### DISCUSSION THREAD

**Discussion by erikhofer** [react-hook-form/react-hook-form#9360] [Q&A] [answered]
I created a reproduction with codesandbox: https://codesandbox.io/s/shy-water-cz9ryj?file=/src/App.js

Depending on how I declare the default value for my select, it is either not displayed or not valid. 

This looks like a problem with the UI library I am using (Ant Design) because it works with the native `<select>` element. I also filed an issue there, but maybe someone here has an idea of what I am doing wrong :)

**Comment by leapful** [comment]
@erikhofer 

Your zod schema is expecting enum value "a", "b" or "c", that why with the default value "a", the functionality is still working. Switching to default value `Foos.a` equivalent to an object `{ key: 'a', label: 'A' }` therefore zod throws a validation error.

```
const Schema = z.object({
  foo: z.enum(["a", "b", "c"]).transform((key) => Foos[key])
});
```

**Reply by erikhofer** [reply]
Okay, so default value "a" is the right way to do it, that is good to know.
Now I just have to make this work with the Ant Design `<Select>`.

**Reply by Butcherktg** [reply]
Yes

**Comment by erikhofer** [comment, marked as answer]
If somebody comes across this, the solution can be found here: https://stackoverflow.com/questions/74428889/how-to-use-ant-design-select-with-react-hook-forms-and-value-transformation
